### PR TITLE
Add Example for lockable Tooltip

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description

Create a story that shows a Tooltip, where clicking on the chart fixes the Tooltip content and its position. Clicking it again, unlocks the Tooltip. 

I worked on this and wanted to share it :)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/Examples/Tooltip.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+export const LockedByClick = {
+  render: (_args: Record<string, any>) => {
+    const [isLocked, setIsLocked] = React.useState(false);
+    // The TooltipData contains the payload, the label and the x position of the tooltip.
+    // Their update is interrupted by the click event, so we need to store them in a state.
+    const [tooltipData, setTooltipData] = React.useState<{ payload?: unknown[]; label?: string; x?: number }>({});
+
+    // A custom Tooltip that updates the payload of the tooltip if the chart is locked, and either way always renders using the normal Tooltip.
+    const CustomTooltip = (props: any) => {
```
Comment (ckifer) [inline storybook/stories/Examples/Tooltip.stories.tsx:19]: we know at least somewhat what the type of CustomTooltip is? Is it possible to partially type this?

**Review by ckifer — APPROVED** [review]
Cool! That if `CustomTooltip` was in a different file and outside of the scope of `setTooltipData`? Would you pass the setter in? Or to keep it more generic maybe provide a callback function to set the data?

Nothing blocking, just thinking aloud. Thanks!

**Comment by ckifer** [comment]
Nothing blocking, merging in :)

**Comment by nikolasrieble** [comment]
> Cool! What if `CustomTooltip` was in a different file and outside of the scope of `setTooltipData`? Would you pass the setter in? Or to keep it more generic maybe provide a callback function to set the data?
> 
> Nothing blocking, just thinking aloud. Thanks!

Good questions! I admittedly did not think about it yet. Passing the setter seems like a reasonable suggestion.

**Pull request merged**